### PR TITLE
[TIMOB-23620] Add device pair compatibility for Xcode 8

### PR DESCRIPTION
**Pull request by janvennemann**
JIRA: https://jira.appcelerator.org/browse/TIMOB-23620

I adopted the pairings Xcode allows you to make within its Device Manager

### CODE REVIEW

File: lib/simulator.js [hunk 1]
```diff
@@ -92,6 +92,15 @@ const devicePairCompatiblity = exports.devicePairCompatiblity = {
 		'9.x': {        // iOS 9.0
 			'2.x': true // watchOS 2.0
 		}
+	},
+	'8.x': {
```
Comment (cb1kenobi) [inline lib/simulator.js]: I wonder if we should set this to `>=8.x` to future proof us?

Comment (janvennemann) [inline lib/simulator.js]: Xcode typically includes the simulators for the current iOS sdk version and the previous version sims can be downloaded. If Xcode 9 come out in the future, it probably supports iOS 11 and iOS 10 sims. So we would have to edit this mapping anyway.

File: lib/simulator.js [hunk 2]
```diff
@@ -92,6 +92,15 @@ const devicePairCompatiblity = exports.devicePairCompatiblity = {
...
+		'>=9.0 <=9.3': {
```
Comment (cb1kenobi) [inline lib/simulator.js]: This should be `9.x`, not a range.

File: lib/simulator.js [hunk 3]
```diff
@@ -92,6 +92,15 @@ const devicePairCompatiblity = exports.devicePairCompatiblity = {
...
+			'2.x': true
+		},
```
Comment (cb1kenobi) [inline lib/simulator.js]: This is not correct. You CAN pair a WatchOS 3 simulator with a iOS 9 simulator. Please add `'3.x': true`.

**Comment by janvennemann** [comment]
As you suggested, i changed the iOS 9 constraint to 9.x and enabled watchOS 3 support for those sims.

**Comment by cb1kenobi** [comment]
@janvennemann Thanks, testing now. :)

**Comment by cb1kenobi** [comment]
Looks good. **APPROVED**

**Pull request merged**